src/main/java/seedu/address/logic/commands/LinkLoanCommand.java

```diff
@@ -0,0 +1,177 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_RETURN_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_START_DATE;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_VALUE;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Links a loan to a person in the address book.
+ */
+public class LinkLoanCommand extends Command {
+
+    public static final String COMMAND_WORD = "link";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Links a loan to the person identified "
+            + "by the index number used in the displayed person list. "
+            + "Parameters: "
+            + "INDEX "
+            + PREFIX_VALUE + "VALUE "
+            + PREFIX_START_DATE + "START_DATE "
+            + PREFIX_RETURN_DATE + "RETURN_DATE\n"
+            + "Example: " + COMMAND_WORD + " "
+            + "5 "
+            + PREFIX_VALUE + "500.00 "
+            + PREFIX_START_DATE + "15-02-2024 "
+            + PREFIX_RETURN_DATE + "21-04-2024";
+
+    public static final String MESSAGE_SUCCESS = "New loan linked: %1$s";
```

Comment:
Do remember to include MESSAGE_FAILURE in future as well

Reply:
Yup, loan linking failure is already handled with an appropriate error message but I will aim to create a custom error message for the `LinkLoanCommand` class as well in the future. Thanks!